frontend/src/react-query/api/agent.tsx

```diff
@@ -169,16 +169,24 @@ export const useListAgentsQuery = (
     refetchInterval?: number | false;
   },
 ) => {
+  const listPipelinesRequestFilter = create(ListPipelinesRequest_FilterSchema, {
+    tags: input?.filter?.nameContains
+      ? {
+          __redpanda_cloud_pipeline_type: 'agent',
+          // Ensure name contains is only used if a string is provided. It has to match the agent name exactly.
+          // We cannot use nameContains because the name of the agent is stored as a tag, not pipeline name.
+          // TODO: Once dedicated "agent service" exists, we can use nameContains for that endpoint to filter by agent name.
+          __redpanda_cloud_agent_name: input?.filter?.nameContains,
```

Comment:
This is not a nameContains filter, the tags only filter on exact matches.

Reply:
Yes, I moved it to a comment because I want to keep using the filter schema from proto